### PR TITLE
WT-12018 Fix metadata info comparison in test_import_base

### PR DESCRIPTION
**Pull request by etienneptl**
Before the suggested changed, we would compare `None` with `None` as `sort()` does not return the sorted list. I have also tried to simplify the information that is unique and that cannot be compared: the `id` and `checkpoint`.

### CODE REVIEW

File: test/suite/test_import01.py [hunk 1]
```diff
@@ -74,16 +74,18 @@ def check(self, uri, keys, values):
...
     def config_compare(self, aconf, bconf):
         a = re.sub('id=\d+,?', '', aconf)
-        a = (re.sub('\w+=\(.*?\)+,?', '', a).strip(',').split(',') +
-             re.findall('\w+=\(.*?\)+', a))
+        a = re.sub('checkpoint=\(.*?\)+,?', '', a)
         b = re.sub('id=\d+,?', '', bconf)
-        b = (re.sub('\w+=\(.*?\)+,?', '', b).strip(',').split(',') +
-             re.findall('\w+=\(.*?\)+', b))
-        self.assertTrue(a.sort() == b.sort())
+        b = re.sub('checkpoint=\(.*?\)+,?', '', b)
```
Comment (korteland) [inline test/suite/test_import01.py:83]: If this is a bug, it would have existed before this change too. But wouldn't this match too far? i.e. given a string `checkpoint=(some,stuff),shouldnt_be_deleted=true,b=(some,other,stuff)` - it would match the entire string rather than just the `checkpoint=(some,stuff)` clause?

Comment (korteland) [reply test/suite/test_import01.py:83]: Hmm, it seems to work in a REPL. I don't know why it's not being "greedy" like I'd expect.

**Review by korteland — APPROVED** [review]

File: test/suite/test_import01.py (hunk 1 above)
Comment (korteland) [reply test/suite/test_import01.py:83]: Oh, it's the `.*?` construct - I get it now.

File: test/suite/test_import01.py [hunk 2]
```diff
@@ -74,16 +74,18 @@ def check(self, uri, keys, values):
             else:
                 self.check_record(uri, keys[i], values[i])
 
-    # We know the ID can be different between configs, so just remove it from comparison.
-    # Everything else should be the same.
+    # The ID and checkpoint information can be different between configs, remove it. Everything else
+    # should be the same.
     def config_compare(self, aconf, bconf):
         a = re.sub('id=\d+,?', '', aconf)
```
Comment (ajmorton) [inline test/suite/test_import01.py:80]: This isn't part of the current ticket, but the `id` regex will also match against the contents of the encryption field `encryption=(keyid=,` if the key starts with a digit. This won't impact correctness since the removal will happen on both configs

File: test/suite/test_import01.py (hunk 1 above)
Comment (ajmorton) [reply test/suite/test_import01.py:83]: I ran `test_import01` and dumped `aconf`. The checkpoint field looks like
```
checkpoint=( 
    WiredTigerCheckpoint.1=( 
        addr="018181e40de2a8838281e488c2b6928383e48aff1d60808080cdc0c1c0",
        order=1,
        time=1700608072,
        size=1024,
        newest_start_durable_ts=60,
        oldest_start_ts=10,
        newest_txn=8,
        newest_stop_durable_ts=0,
        newest_stop_ts=-1,
        newest_stop_txn=-11,
        prepare=0,
        write_gen=3,
        run_write_gen=1
    )
),
```
which means the greedy parsing will miss the final brace. We'll also run into trouble if the checkpoint config contains multiple checkpoint (i.e. named checkpoints). I'll think on this but I'm not sure regex is enough to handle the edge cases

File: test/suite/test_import01.py [hunk 3]
```diff
@@ -74,16 +74,18 @@ def check(self, uri, keys, values):
...
-        a = (re.sub('\w+=\(.*?\)+,?', '', a).strip(',').split(',') +
-             re.findall('\w+=\(.*?\)+', a))
+        a = re.sub('checkpoint=\(.*?\)+,?', '', a)
         b = re.sub('id=\d+,?', '', bconf)
-        b = (re.sub('\w+=\(.*?\)+,?', '', b).strip(',').split(',') +
```
Comment (ajmorton) [inline test/suite/test_import01.py:84]: The previous code was stripping all the subconfigs (ignoring my other comment on greedy regexes), sorting the simple configs, and then appending all the subconfigs at the end before sorting. 
With the new code that doesn't do this we'll end up splitting subconfigs like `version=(major=2,minor=1),` into two items `version=(major` and `minor=1)` for comparison. It's still correct, but seems messy.

**Review by ajmorton — COMMENTED** [review]
Couple of style/cleanliness notes and some future proofing, but the python regex looks good and correctness for our current test suite seems ok. Happy to split my comments out into a separate ticket if this unblocks your project work.

File: test/suite/test_import01.py (hunk 1 above)
Comment (etienneptl) [reply test/suite/test_import01.py]: There was definitely an issue with the regex and having multiple (named) checkpoints. I reverted the regex and now explicitly remove checkpoints not using the regex as you suggested in 0e8742b3083361e195bd313aeafeccf5cf102745.

File: test/suite/test_import01.py (hunk 3 above)
Comment (etienneptl) [reply test/suite/test_import01.py:84]: I agree, I reverted the regex.

**Review by ajmorton — APPROVED** [review]
LGTM 👍

**Review by korteland — APPROVED** [review]
LGTM

**Comment by etienneptl** [comment]
evergreen merge
